ncm-spma/src/main/perl/spma/apt.pm

```diff
@@ -0,0 +1,433 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+package NCM::Component::spma::apt;
+
+=head1 NAME
+
+NCM::Component::spma::apt - NCM SPMA backend for apt
+
+=head1 SYNOPSIS
+
+This document describes how to control the behaviour of the package manager itself.
+For information on how to manage packages with Quattor, please check
+L<http://quattor.org/documentation/2013/04/05/package-management.html>.
+
+=head1 DESCRIPTION
+
+apt.pm implements an apt backend for ncm-spma, the approach taken is to defer as much work as possible to apt.
+
+A single SPMA run consists of the following steps:
+
+=over
+
+=item
+
+Setup repository directory if required
+
+=item
+
+Remove repositories that are not found in the profile
+
+=item
+
+Update repository cache from upstream sources
+
+=item
+
+Upgrade already installed packages
+
+=item
+
+Install packages specified in the profile that are not installed
+
+=item
+
+Mark any packages installed but not in the profile as automatically installed
+
+=item
+
+Ask apt to remove all automatically installed packages that are not satisfying dependencies of other packages
+
+=back
+
+Support for groups (tasks) is not currently provided.
+
+=head1 RESOURCES
+
+Only a very minimal schema is implemented.
+
+Repositories listed under C</software/repositories> will be configured,
+URLs should be followed by the suite and sections required e.g. http://example.org/debian unstable main
+
+Packages listed under C</software/packages> will be installed, version and architecture locking (including multiarch) is fully implemented.
+
+=cut
+
+use strict;
+use warnings;
+use NCM::Component;
+our $EC=LC::Exception::Context->new->will_store_all;
+our @ISA = qw(NCM::Component);
+use EDG::WP4::CCM::Element qw(escape unescape);
```

Comment:
once https://github.com/quattor/CCM/pull/102 is merged, you should import the escape functions from `CCM::Path` (altough current code will still work)